graph/path/dijkstra_test.go

```diff
@@ -56,38 +79,41 @@ func TestDijkstraFrom(t *testing.T) {
 				t.Fatalf("%q %s: unexpected from node ID: got:%d want:%d", test.Name, tg.typ, pt.From().ID(), test.Query.From().ID())
 			}
 
-			p, weight := pt.To(test.Query.To().ID())
-			if weight != test.Weight {
-				t.Errorf("%q %s: unexpected weight from To: got:%f want:%f",
-					test.Name, tg.typ, weight, test.Weight)
-			}
 			if weight := pt.WeightTo(test.Query.To().ID()); weight != test.Weight {
 				t.Errorf("%q %s: unexpected weight from Weight: got:%f want:%f",
 					test.Name, tg.typ, weight, test.Weight)
 			}
 
-			var got []int64
-			for _, n := range p {
-				got = append(got, n.ID())
-			}
-			ok := len(got) == 0 && len(test.WantPaths) == 0
-			for _, sp := range test.WantPaths {
-				if reflect.DeepEqual(got, sp) {
-					ok = true
-					break
-				}
-			}
-			if !ok {
-				t.Errorf("%q %s: unexpected shortest path:\ngot: %v\nwant from:%v",
-					test.Name, tg.typ, p, test.WantPaths)
-			}
+			p, weight := pt.To(test.Query.To().ID())
+			checkPath(tg.typ, p, weight)
 
 			np, weight := pt.To(test.NoPathFor.To().ID())
 			if pt.From().ID() == test.NoPathFor.From().ID() && (np != nil || !math.IsInf(weight, 1)) {
 				t.Errorf("%q %s: unexpected path:\ngot: path=%v weight=%f\nwant:path=<nil> weight=+Inf",
 					test.Name, tg.typ, np, weight)
 			}
 		}
+
```

Comment:
I think this should be in the loop, with helper sharing.

Reply:
Done.